datalad/metadata/search.py

```diff
@@ -181,15 +181,14 @@ def _get_search_schema(ds):
                 # might be different when some aggregated metadata was
                 # generated with an old version of datalad
                 # in this case we should actually load the old vocabulary
-                set.add(', '.join(i for i in v) if isinstance(v, (tuple, list)) else v)
+                #set.add(', '.join(i for i in v) if isinstance(v, (tuple, list)) else v)
                 # and perform the mapping to the current one in here
                 count = 0
                 uk = k
                 while uk in definitions:
                     count += 1
                     uk = '{}_{}'.format(k, count)
-                ds_defs[k] = uk
-                k = uk
+                ds_defs[k] = k = uk
```

Comment:
TBH: I don't like this kind of diff. It is functionally irrelevant and requires more thinking when reading (are you really sure you know the order of evaluation in compound code like this all the time?).